rundeckapp/grails-app/assets/javascripts/user/profile.js

```diff
@@ -142,7 +142,8 @@ function generateUserToken(login, elem, data) {
         beforeSend: _createAjaxSendTokensHandler('api_req_tokens'),
         success: function (data, status, jqxhr) {
             if (data.result) {
-                addTokenRow(elem, login, data.tokenid);
+              // Reload current page.
+              window.location.reload();
```

Comment:
Is there a way to detect if the adding a token to the row would push it over the page max and only then do a reload? The user interaction is less pleasant with the reload on every token creation. I'm guessing having over 20 tokens is a corner case anyway, so if we could only reload when absolutely necessary that would be great.

Reply:
Ok i adjusted it so it will only will reload if the insertion will overflow the table beyond the max page length.